envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoyEgressRoutesFactory.kt

```diff
@@ -120,6 +121,10 @@ internal class EnvoyEgressRoutesFactory(
             routeAction.setInternalRedirectAction(RouteAction.InternalRedirectAction.HANDLE_INTERNAL_REDIRECT)
         }
 
+        routeAction
```

Comment:
We should add some property which enable this feature. Also a name of the header should be configurable. I think we don't want to do this for all services connected to control plane. Maybe this information should be added in Envoy metadata? WDYT @MarcinFalkowski @slonka @pbetkier

Reply:
https://github.com/allegro/envoy-control/pull/96#discussion_r396493669